base/indices.jl

```diff
@@ -13,7 +13,30 @@ Indices{N} = NTuple{N,AbstractUnitRange}
 ## Traits for array types ##
 
 abstract type IndexStyle end
+"""
+    IndexLinear()
+
+Subtype of [`IndexStyle`](@ref) which specifies that the array it describes
```

Comment:
Really nice.  Here's a shot at a set of hopefully-slightly-clearer first sentences:

> Subtype of ``[`IndexStyle`](@ref)`` that is used to describe arrays that are optimally indexed by one **linear** index.

> Subtype of ``[`IndexStyle`](@ref)`` that is used to describe arrays that are optimally indexed by a **Cartesian** index with the same number of indices as there are dimensions in the array.

I also have a slight preference for separating out the first sentence in a docstring as a stand-alone "summary" line (a paragraph by itself).